tests/test_recipe.py

```diff
@@ -112,6 +112,77 @@ def test_standard_recipe_with_balancer(self):
 
         self.assertEqual(counts["entailment"], counts["not entailment"])
 
+    def test_standard_recipe_with_loader_limit(self):
+        recipe = StandardRecipeWithIndexes(
+            card="cards.wnli",
+            instruction="instructions.models.llama",
+            template="templates.key_val",
+            format="formats.user_agent",
+            demos_pool_size=5,
+            num_demos=1,
+            loader_limit=10,
+        )
+
+        stream = recipe()
+        self.assertEqual(len(list(stream["train"])), 5)
```

Comment:
this is confusing, why only 5 and not 10 also?

Reply:
Because 5 elements are taken to the demo pool.  Added comment.